adapters/rediads/rediads.go

```diff
@@ -0,0 +1,157 @@
+
+import (
+	"fmt"
+	"github.com/buger/jsonparser"
+	"github.com/prebid/openrtb/v20/openrtb2"
+	"github.com/prebid/prebid-server/v3/adapters"
+	"github.com/prebid/prebid-server/v3/config"
+	"github.com/prebid/prebid-server/v3/errortypes"
+	"github.com/prebid/prebid-server/v3/openrtb_ext"
+	"github.com/prebid/prebid-server/v3/util/jsonutil"
+)
+
+type adapter struct {
+	endpoint string
+}
+
+// Builder builds a new instance of the adapter for the given bidder with the given config.
+func Builder(bidderName openrtb_ext.BidderName, config config.Adapter, server config.Server) (adapters.Bidder, error) {
+	bidder := &adapter{
+		endpoint: config.Endpoint,
+	}
+	return bidder, nil
+}
+
+func (a *adapter) MakeRequests(request *openrtb2.BidRequest, requestInfo *adapters.ExtraRequestInfo) ([]*adapters.RequestData, []error) {
+	var errors []error
+	var accountID string
+	var endpoint string
+
+	// Iterate through all impressions in the request
+	for i, imp := range request.Imp {
+		// Extract and validate bidder-specific params from imp.Ext
+		var bidderExt adapters.ExtImpBidder
+		var rediadsExt openrtb_ext.ExtImpRediads
+
+		// Unmarshal bidder extension
+		if err := jsonutil.Unmarshal(imp.Ext, &bidderExt); err != nil {
+			errors = append(errors, &errortypes.BadInput{
+				Message: fmt.Sprintf("Invalid Ext format in impression %s", imp.ID),
+			})
+			continue
+		}
+		// Remove "prebid" and "bidder" from imp.Ext
+		newExt := jsonparser.Delete(imp.Ext, "prebid")
+		newExt = jsonparser.Delete(newExt, "bidder")
+		imp.Ext = newExt
+
+		// Unmarshal custom bidder params
+		if err := jsonutil.Unmarshal(bidderExt.Bidder, &rediadsExt); err != nil {
+			errors = append(errors, &errortypes.BadInput{
+				Message: fmt.Sprintf("Invalid bidder params in impression %s", imp.ID),
+			})
+			continue
+		}
+
+		// Set accountID and slot to use in publisher id and imp tag id respectively
+		accountID = rediadsExt.AccountID
+		endpoint = rediadsExt.Endpoint
+
+		// Set tagid in the imp object
+		if rediadsExt.Slot != "" {
+			imp.TagID = rediadsExt.Slot
+		}
+		request.Imp[i] = imp
+	}
+
+	// Update site or app publisher ID with accountID
+	if request.Site != nil {
+		siteCopy := *request.Site
+		if siteCopy.Publisher != nil {
+			publisherCopy := *siteCopy.Publisher
+			publisherCopy.ID = accountID
+			siteCopy.Publisher = &publisherCopy
+		} else {
+			siteCopy.Publisher = &openrtb2.Publisher{ID: accountID}
+		}
+		request.Site = &siteCopy
+	} else if request.App != nil {
+		appCopy := *request.App
+		if appCopy.Publisher != nil {
+			publisherCopy := *appCopy.Publisher
+			publisherCopy.ID = accountID
+			appCopy.Publisher = &publisherCopy
+		} else {
+			appCopy.Publisher = &openrtb2.Publisher{ID: accountID}
+		}
+		request.App = &appCopy
+	}
+
+	// Marshal the final request
+	requestJSON, err := jsonutil.Marshal(request)
+	if err != nil {
+		return nil, []error{err}
+	}
+
+	finalEndpoint := a.endpoint
+	if endpoint != "" {
+		finalEndpoint = endpoint
```

Comment:
Need a json test case here for setting the endpoint based on the endpoint passed in from imp.ext